metals/src/main/scala/scala/meta/internal/metals/ForwardingMetalsBuildClient.scala

```diff
@@ -66,13 +68,24 @@ final class ForwardingMetalsBuildClient(
   ): List[LogForwarder] = {
     forwarders.getAndUpdate(_.prepended(logForwarder))
   }
-  private case class Compilation(
-      timer: Timer,
-      promise: Promise[b.CompileReport],
-      isNoOp: Boolean,
-      progress: TaskProgress = TaskProgress.empty,
+  private class Compilation(
+      val timer: Timer,
+      val isNoOp: Boolean,
+      token: Option[Future[WorkDoneProgress.Token]],
```

Comment:
```suggestion
      token: Option[WorkDoneProgress.Token],
```
could we avoid having a future here? We could probably change startProgress to be generic?

Reply:
It's just to make sure that progress was actually created on client side before we send e.g. `end`.
> We could probably change startProgress to be generic?
I'm not sure I understand.

Reply:
> I'm not sure I understand.

I don't remember what I had in mind :sweat_smile: 

Maybe we could have a sealed hierachy instead, soemthing like:
```
sealed trait Compilation{
  def timer: Timer
  def taskProgress: TaskProgress
}

case class NoopCopilation(
  val timer: Timer,
){
  override def taskProgress: TaskProgress = TaskProgress.empty,
}

case class IncrementalCompilation(  
      val timer: Timer,
      token: Future[WorkDoneProgress.Token],
       taskProgress: TaskProgress
)
```
what do you think?

Reply:
I think that after @ckipp01's changes in Bloop we need a `noop`  anymore, Bloop won't send `startTask` for it anymore.